src/main/java/ch/njol/skript/classes/data/DefaultOperations.java

```diff
@@ -99,21 +99,24 @@ public class DefaultOperations {
 		// Timespan - Number
 		// Number - Timespan
 		Arithmetics.registerOperation(Operator.MULTIPLICATION, Timespan.class, Number.class, (left, right) -> {
-			long scalar = right.longValue();
-			if (scalar < 0)
+			double scalar = right.doubleValue();
+			if (scalar < 0 || !Double.isFinite(scalar))
 				return null;
-			return new Timespan(Math2.multiplyClamped(left.getAs(TimePeriod.MILLISECOND), scalar));
+			double value = left.getAs(TimePeriod.MILLISECOND) * scalar;
+			return new Timespan((long) Math.min(value, Long.MAX_VALUE));
 		}, (left, right) -> {
-			long scalar = left.longValue();
-			if (scalar < 0)
+			double scalar = left.doubleValue();
+			if (scalar < 0 || !Double.isFinite(scalar))
 				return null;
-			return new Timespan(scalar * right.getAs(TimePeriod.MILLISECOND));
+			double value = right.getAs(TimePeriod.MILLISECOND) * scalar;
+			return new Timespan((long) Math.min(value, Long.MAX_VALUE));
 		});
 		Arithmetics.registerOperation(Operator.DIVISION, Timespan.class, Number.class, (left, right) -> {
-			long scalar = right.longValue();
-			if (scalar <= 0)
+			double scalar = right.doubleValue();
+			if (scalar < 0 || !Double.isFinite(scalar))
 				return null;
```

Comment:
```suggestion
			if (scalar <= 0 || !Double.isFinite(scalar))
				return null;
```
i think this should be done aswell

Reply:
Oh that is true I misread the original code. What do others think?

Reply:
It makes sense but I can't think of a case where I would want `timespan / 0` to be the eternity

Reply:
given that 1/0 returns infinity, this should probably also return infinity

Reply:
It cannot, though. Infinity isn't a timespan, and a timespan divided by a number is a timespan.

Reply:
I'm going to revert to previous behavior since there seems to be no strong feelings.